src/lib/validators.js

```diff
@@ -0,0 +1,108 @@
+import { keccak256 } from 'eth-lib/lib/hash.js';
+
+export const ethValidators = {
+  isHex(value) {
+    const v = String(value);
+    if (v.length < 2) {
+      return false;
+    }
+    const prefix = v.slice(0, 2).toLowerCase();
+    if (prefix != '0x') {
+      return false;
+    }
+    const hex = v.slice(2);
+    if (!/^[0-9a-f]*$/i.test(hex)) {
+      return false;
+    }
+    return true;
+  },
+  isAddr(value) {
+    if (!ethValidators.isHex(value)) {
+      return false;
+    }
+    const hex = String(value).slice(2);
+    if (!/^[0-9a-f]{40}$/i.test(hex)) {
+      return false;
+    }
+    return true;
+  },
+  isChecksummed(value) {
+    if (!ethValidators.isAddr(value)) {
+      return false;
+    }
+    var address = String(value).slice(2);
+    var addrLC = address.toLowerCase();
+    var addrUC = address.toUpperCase();
+    var addrHash = keccak256(addrLC).replace(/^0x/i, '');
+    for (var i = 0; i < 40; i++) {
+      // the nth letter should be uppercase if the nth digit of casemap is 1
+      if (parseInt(addrHash[i], 16) > 7) {
+        if (addrUC[i] !== address[i]) {
+          return false;
+        }
+      } else {
+        if (addrLC[i] !== address[i]) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+};
+
+export function validateTx(tx) {
+  let errors = [];
+
+  // Validate: to
+  if (!ethValidators.isHex(tx.to)) {
+    errors.push('To is not valid hex');
+  }
+  if (!ethValidators.isAddr(tx.to)) {
+    errors.push('To is not an Ethereum address');
+  }
+  if (!ethValidators.isChecksummed(tx.to)) {
+    errors.push('To has an incorrect checksum');
+  }
+
+  // Validate: from
+  if (tx.from) {
```

Comment:
Looks odd. `to` is optional, `from` is required. Looks like you maybe implemented it the other way around?

Reply:
oops, you're right